js/client.js

```diff
@@ -1011,6 +1021,72 @@
 				break;
 			}
 		},
+		parseGroups: function (groupsList) {
+			var data = null;
+			try {
+				data = JSON.parse(groupsList);
+			} catch (e) {}
+			if (!data) return; // broken JSON - keep default ranks
+
+			var authTiers = [[], [], []]; // we will be splitting them into three tiers, before compiling them back into the main tree
+
+			// process the data and sort into the three auth tiers, 0, 1, and 2
+			for (var i = 0; i < data.length; i++) {
+				var entry = data[i];
+				if (!entry) continue;
+
+				var symbol = entry.symbol || ' ';
+				var groupName = entry.name;
+				var groupType = entry.type || 0;
+
+				if (!groupName) this.defaultGroup = symbol;
+
+				authTiers[groupType].push({
+					symbol: symbol,
+					data: {
+						name: groupName ? Tools.escapeHTML(groupName + ' (' + symbol + ')') : null,
+						type: groupType,
+					},
+				});
+			}
+
+			// add the 3 last groups: muted, namelocked and locked
+			authTiers[0] = authTiers[0].concat([
+				{
+					symbol: '!',
+					data: {
+						name: "<span style='color:#777777'>Muted (!)</span>",
+						type: 0,
+					},
+				},
+				{
+					symbol: '✖',
+					data: {
+						name: "<span style='color:#777777'>Namelocked (✖)</span>",
```

Comment:
Also, maybe these three should also be sent by the server. Type `-1` or something.